app/serializers/vet_payment_history_serializer.rb

```diff
@@ -10,11 +10,63 @@ def id
     nil
   end
 
+  def initialize(object, options = nil)
+    @formatted_payments = []
+    @returned_payments = []
+    process_payments(object[:payments][:payment]) if object[:payments].present?
+    super
+  end
+
   def payments
-    object[:payments]
+    @formatted_payments
   end
 
   def return_payments
-    object[:return_payments]
+    @returned_payments
+  end
+
+  private
+
+  # rubocop:disable Metrics/MethodLength
+  def process_payments(all_payments)
+    all_payments.each do |payment|
+      if payment.dig(:return_payment, :check_trace_number).present?
+        @returned_payments << {
+          'returned_check_issue_dt': payment[:payment_date],
+          'returned_check_cancel_dt': payment[:return_payment][:return_date],
+          'returned_check_amount': ActiveSupport::NumberHelper.number_to_currency(payment[:payment_amount]),
+          'returned_check_number': payment[:return_payment][:check_trace_number],
+          'returned_check_type': payment[:payment_type],
+          'return_reason': payment[:return_payment][:return_reason]
+        }
```

Comment:
I forget if we tried this or not but in case not, I wonder if we can pull this hash and the one below into their own methods 🤔 . 

If we do, I bet we can drop the cop disable.

Reply:
Thanks!  I've moved the return payments into a separate method.